.github/workflows/runAqaArgParse.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import json
+import sys
+
+def map_platforms(platforms):
+    """ Takes in a list of platforms and translates Grinder platorms to corresponding GitHub-hosted runners.
+        This function both modifies and returns the 'platforms' argument.
+    """
+    
+    platform_map = {
+        'x86-64_windows': 'windows-latest',
+        'x86-64_mac': 'mac-latest',
+        'x86-64_linux': 'ubuntu-latest'
+    }
+    
+    for i, platform in enumerate(platforms):
+        if platform in platform_map:
+            platforms[i] = platform_map[platform]
+    
+    return platforms
+
+def underscore_targets(targets):
+    """ Takes in a list of targets and prefixes them with an underscore if they do not have one.
+    """
+    result = []
+    for target in targets:
+        t = target
+        if target[0] != '_':
+            t = '_' + t
+        result.append(t)
+    return result
+
+def main():
+
+    # The keyword for this command.
+    keyword = 'run aqa'
+    keywords = keyword.split()
+
+    # We assume that the first argument is/are the keyword(s).
+    # e.g. sys.argv == ['action_argparse.py', 'run', 'aqa', ...]
+    raw_args = sys.argv[1 + len(keywords):]
+
+    parser = argparse.ArgumentParser(prog=keyword, add_help=False)
+    # Improvement: Automatically resolve the valid choices for each argument populate them below, rather than hard-coding choices.
+    parser.add_argument('--sdk_resource', default=['nightly'], choices=['nightly', 'releases'], nargs='+')
+    parser.add_argument('--build_list', default=['openjdk'], choices=['openjdk', 'functional', 'system', 'perf', 'external'], nargs='+')
+    parser.add_argument('--target', default=['_jdk_math'], nargs='+')
+    parser.add_argument('--platform', default=['x86-64_linux'], nargs='+')
+    parser.add_argument('--jdk_version', default=['8'], nargs='+')
+    parser.add_argument('--jdk_impl', default=['openj9'], nargs='+')
```

Comment:
Would it be nicer to set jdk_impl and platform as choices, similarly to build_list and sdk_source?

Reply:
It would be `openj9` and `hotspot` as the only two choices?

Reply:
> It would be openj9 and hotspot as the only two choices?  For jdk_impl @smlambert

Reply:
yes those 2 choices (and should we ever need it, in future, we can add jdk_vendor as an argument, but will not include it for now)

Reply:
I have set platform as 'x86-64_linux ` with extra space by accident and got https://github.com/sophia-guo/openjdk-tests/runs/1919706546?check_suite_focus=true. It would be nice to set platform as choices so the job would fail when do the parseComment instead of runBuild and can't find the runner.

Reply:
> I have set platform as 'x86-64_linux ` with extra space by accident and got https://github.com/sophia-guo/openjdk-tests/runs/1919706546?check_suite_focus=true. It would be nice to set platform as choices so the job would fail when do the parseComment instead of runBuild and can't find the runner.

That would be quite a long choices list. The choice allows github runner platforms, which include various versions of windows, mac, and linux.

Reply:
`platform` is what jdk supports on. Here is the list https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/buildenv/jenkins/openjdk_tests. `windows-latest` or others are supported virtual-environments in github runners. https://github.com/actions/virtual-environments. 

That is no all supported platforms  are enabled on github runners.
There is a map between them. 

For example `x86-64_linux`, `x86-64_linux_mixed`, `x86-64_linux_xl` can map to any one of `ubuntu-16.04`, `ubuntu-latest or ubuntu-18.04`, `ubuntu-16.04`. 

So choices would be [`x86-64_linux`, `x86-64_linux_mixed`, `x86-64_linux_xl`, `x86-64_mac` , `x86-64_mac_mixed`, `x86-64_mac_xl`, `x86-64_windows`,`x86-64_windows_mixed`,`x86-64_windows_xl`]. Going forward some of them will be removed or combined so the list would be smaller. 

We can leave it as is and update later if the choices helps.